src/services/new/fileUploadManger.ts

```diff
@@ -0,0 +1,305 @@
+          // don't need to check verifiction result here because we know it was just signed
+          await bls.circuit.addAndVerify(sigData.p, sigData.s)
+          responses.set(cid.toString(), bls)
+          await ipfs.pubsub.publish(
+            `${config.get('network.id')}-${RESPONSE_TOPIC}`,
+            encoder.encode(JSON.stringify(resp)),
+          )
+        } else {
+          const resp: FileUploadResponseInfo = {
+            type: 'error',
+            // @ts-ignore cid is not never here idk what ts is talking about
+            cid: request.cid.toString(),
+            error: 'data does not match cid',
+          }
+          await ipfs.pubsub.publish(
+            `${config.get('network.id')}-${RESPONSE_TOPIC}`,
+            encoder.encode(JSON.stringify(resp)),
+          )
+        }
+
+        // remove the request from the queue
+        clearTimeout(request.autoRemoveTimeout)
+        requests.delete(msg.from.toString())
+      }),
+    )
+
+    // Step 3 - Coordinate with other nodes to generate a valid signed response
+    await ipfs.pubsub.subscribe(
+      `${config.get('network.id')}-${RESPONSE_TOPIC}`,
+      ignoreUnsignedMessages(
+        ignoreMessagesFromSelf(ipfs, async (msg) => {
+          const respInfo: FileUploadResponseInfo = parseFileUploadResponseInfo(
+            decoder.decode(msg.data),
+          )
+
+          if (respInfo.type === 'error') {
+            return
+          }
+
+          if (respInfo.type === 'proof') {
+            const msg = CID.parse(respInfo.data).bytes
+            const proof = BlsCircuit.deserialize(
+              {
+                hash: msg,
+                signature: respInfo.signature,
+              },
+              multisig.circuitMap,
+            )
+            if (proof.aggPubKeys.size >= REQUIRED_VERIFIERS && (await proof.verify(msg))) {
+              // got a valid proof stop trying to generate a new one
+              responses.delete(respInfo.cid)
+            }
+            return
+          }
+
+          const bls = responses.get(respInfo.cid)
+
+          if (!bls) {
+            return
+          }
+
+          // if a valid signature already exists for this cid
+          // then ignore this message
+          const circuit = bls.circuit.finalize()
+          if (circuit.aggPubKeys.size >= REQUIRED_VERIFIERS) {
+            return
+          }
+
+          // validate signature
+          // if invalid then ignore this message
+          const valid = bls.circuit.addAndVerify(respInfo.signature.p, respInfo.signature.s)
+          if (!valid) {
+            return
+          }
+
+          if (circuit.aggPubKeys.size >= REQUIRED_VERIFIERS) {
+            const sig = circuit.serialize(bls.circuit.circuitMap)
+            const data = await ipfs.dag.put(
+              {
+                type: 'data-availablity',
+                cid: respInfo.cid,
+              },
+              { onlyHash: true },
+            )
```

Comment:
Signing CID bytes doesn't indicate the intent of the signature. It could be a number of things. This way makes it clear this is for the data availability proof.